src/App.js

```diff
@@ -1,102 +1,134 @@
+  componentDidMount() {
+    this.hydrateStateWithLocalStorage();
 
-  constructor(prop) {
-    super(prop)
-    this.state = {
-      todoList: todoTest,
-      todoName: '',
-    };
+    // add event listener to save state to localStorage
+    // when user leaves/refreshes the page
+    window.addEventListener(
+      "beforeunload",
+      this.saveStateToLocalStorage.bind(this)
+    );
   }
-  changeHandler = (event) => {
-    this.setState({
-      todoName: event.target.value,
-    });
+
+  componentWillUnmount() {
+    window.removeEventListener(
+      "beforeunload",
+      this.saveStateToLocalStorage.bind(this)
+    );
+
+    // saves if component has a chance to unmount
+    this.saveStateToLocalStorage();
   }
 
-  addTodo = () => {
-    const newTodo = {
-      id: Date.now(),
-      task: this.state.todoName,
-      completed: true,
-    };
+  hydrateStateWithLocalStorage() {
+    // for all items in state
+    for (let key in this.state) {
+      // if the key exists in localStorage
+      if (localStorage.hasOwnProperty(key)) {
+        // get the key's value from localStorage
+        let value = localStorage.getItem(key);
 
-    const newTodoList = this.state.todoList.concat(newTodo);
+        // parse the localStorage string and setState
+        try {
+          value = JSON.parse(value);
+          this.setState({ [key]: value });
+        } catch (e) {
+          // handle empty string
+          this.setState({ [key]: value });
+        }
+      }
+    }
+  }
+  saveStateToLocalStorage() {
+    // for every item in React state
+    for (let key in this.state) {
+      // save to localStorage
+      localStorage.setItem(key, JSON.stringify(this.state[key]));
+    }
+  }
 
+  //Check complete and put line through from CSS
+  markComplete = (id) => {
     this.setState({
-      todoList: newTodoList,
-      todoName: '',
-    });
+      todos: this.state.todos.map(todo => {
+        if (todo.id === id) {
+        todo.completed = !todo.completed
+        }
+        return todo
+    })})
   }
+// Add a todo!
+  addTodo = (title) => {
+    const newTodo = {
+      id: Date.now(),
+      title,
+      completed: false
+    }
+    this.setState({ todos: [...this.state.todos, newTodo] });
+}
 
-  removeComplete = completed => {
-      const completedTasks = this.state.todoList.filter(fr =>  this.state.completed === true);
-      this.setState({
-        todoList: completedTasks,
-      });
+// Check if deleting correct todo by Id
+  delTodo = (id) => {
+    this.setState({todos: [...this.state.todos.filter(todo => todo.id !== id)]})
+  }
+  // Clear all completed todos at once
+  clearComplete = () => {
+    this.setState({todos: [...this.state.todos.filter(todo => todo.completed !== true)]})
   }
-  render() {
+  search = event => {
+    console.log(event.target.value)
+    this.setState({ todo: [...this.state.todos.filter(todo => event.target.value !== todo.title )]})
```

Comment:
If you managed to get the search functionality working don't forget to push this up, please. 👍